typescript-selenium/tests/HappyPath.spec.ts

```diff
@@ -0,0 +1,165 @@
+
+const driver: Driver = e2eContainer.get<Driver>(TYPES.Driver);
+const driverHelper: DriverHelper = e2eContainer.get(CLASSES.DriverHelper);
+const loginPage: LoginPage = e2eContainer.get<LoginPage>(TYPES.LoginPage);
+const dashboard: Dashboard = e2eContainer.get(CLASSES.Dashboard);
+const workspaces: Workspaces = e2eContainer.get(CLASSES.Workspaces);
+const newWorkspace: NewWorkspace = e2eContainer.get(CLASSES.NewWorkspace);
+const workspaceDetails: WorkspaceDetails = e2eContainer.get(CLASSES.WorkspaceDetails);
+const workspaceDetailsPlugins: WorkspaceDetailsPlugins = e2eContainer.get(CLASSES.WorkspaceDetailsPlugins)
+const ide: Ide = e2eContainer.get(CLASSES.Ide)
+const projectTree: ProjectTree = e2eContainer.get(CLASSES.ProjectTree)
+const editor: Editor = e2eContainer.get(CLASSES.Editor)
+
+
+suite("E2E", async () => {
+
+    suite("Login and wait dashboard", async () => {
+        test("login", async () => {
+            await loginPage.login()
+        })
+
+        test("wait dashboard", async () => {
+            await dashboard.waitLoader()
+            await dashboard.waitLoaderDisappearance()
+            await dashboard.waitPage()
+        })
+    })
+
+    suite("Create workspace and open IDE", async () => {
+
+        test("Go to 'New Workspace' page", async () => {
+            await dashboard.clickWorkspacesButton()
+            await workspaces.clickAddWorkspaceButton()
+        })
+
+        test(`Create a '${workspaceName}' workspace`, async () => {
+            await newWorkspace.typeWorkspaceName(workspaceName)
+            await newWorkspace.clickOnChe7Stack()
+            await newWorkspace.waitChe7StackSelected()
+            await newWorkspace.clickOnAddOrImportProjectButton()
+            await newWorkspace.enableSampleCheckbox(sampleName)
+            await newWorkspace.clickOnAddButton()
+            await newWorkspace.waitProjectAdding(sampleName)
+
+            await newWorkspace.selectCreateWorkspaceAndProceedEditing()
+        })
+
+        test("Add 'Java Language Support' plugin to workspace", async () => {
+            const javaPluginName: string = "Language Support for Java(TM)";
+            const execPlugin: string = "Che machine-exec Service";
+
+            await workspaceDetails.waitPage(workspaceName);
+            await workspaceDetails.waitTabSelected('Overview')
+            await workspaceDetails.clickOnTab('Plugins')
+            await workspaceDetails.waitTabSelected('Plugins')
+
+
+            await workspaceDetailsPlugins.waitPluginEnabling(execPlugin)
+            await workspaceDetailsPlugins.waitPluginDisabling(javaPluginName)
+            await workspaceDetailsPlugins.clickOnPluginListItemSwitcher(javaPluginName)
+            await workspaceDetailsPlugins.waitPluginEnabling(javaPluginName)
+
+            await workspaceDetails.waitSaveButton()
+            await workspaceDetails.clickOnSaveButton()
+            await workspaceDetails.waitSaveButtonDisappearance()
+
```

Comment:
Why does the wait need to be in the test? I would prefer to call only one method on page object clickOnSaveButton() and the waits should be in that method.